### PR TITLE
Add basic packed integer arithmetic

### PR DESCRIPTION
**Pull request by jlb6740**

### CODE REVIEW

File: cranelift/codegen/src/isa/x64/lower.rs [hunk 1]
```diff
@@ -47,12 +47,25 @@ fn is_bool_ty(ty: Type) -> bool {
...
+        | types::I32X4
+        | types::I64X2 => false,
         types::I64 => true,
         _ => panic!("type {} is none of I8, I16, I32 or I64", ty),
     }
 }
 
+fn ty_is_packed(ty: Type) -> bool {
+    match ty {
+        types::I8X16 | types::I16X8 | types::I32X4 | types::I64X2 => true,
```
Comment (bjorn3) [inline cranelift/codegen/src/isa/x64/lower.rs:64]: You could simplify this to `ty.lane_count() > 1`.

Comment (jlb6740) [reply cranelift/codegen/src/isa/x64/lower.rs:64]: @bjorn3  Thanks ... will update this.

Comment (abrown) [reply cranelift/codegen/src/isa/x64/lower.rs:64]: I've been using `ty.is_vector() && ty.bits() == 128` on the FP SIMD side.

Comment (jlb6740) [reply cranelift/codegen/src/isa/x64/lower.rs]: 👍 Updated

File: cranelift/codegen/src/isa/x64/lower.rs [hunk 2]
```diff
@@ -47,7 +47,13 @@ fn is_bool_ty(ty: Type) -> bool {
 
 fn int_ty_is_64(ty: Type) -> bool {
     match ty {
-        types::I8 | types::I16 | types::I32 => false,
+        types::I8
+        | types::I16
+        | types::I32
+        | types::I8X16
```
Comment (cfallin) [inline cranelift/codegen/src/isa/x64/lower.rs:53]: This is a bit surprising -- it looks like below, `int_ty_is_64` is only called in the non-vector case, and the panic message below also implies that it's only really applicable to the scalar case. I might be missing something though?

Comment (jlb6740) [reply cranelift/codegen/src/isa/x64/lower.rs:53]: Ahh yes. I didn't want to add these explicitly here. I am thinking we should just: 

types::I64 => true,
_ => panic!("type {} is not I64");

Comment (jlb6740) [reply cranelift/codegen/src/isa/x64/lower.rs]: I think the intent of the message is to catch if the type isn't int, but I don't know how important that is if the function is only called from one place and is this straight forward.

Comment (cfallin) [reply cranelift/codegen/src/isa/x64/lower.rs]: Hmm, yeah, it did serve as a sanity-check (that e.g. we don't have a float type here), but it seems we don't consistently verify types everywhere, and in any case such a problem would be caught by both (i) the CLIF verifier and (ii) asserts/panics at instruction-emission time. So if you want to simply replace it with `ty.unwrap() == types::I64` at the one callsite, I think that's fine.

**Pull request merged**